code/modules/mob/inventory.dm

```diff
@@ -230,8 +230,11 @@
 	return put_in_hand(active_hand, W)
 
 //Puts the item into our inactive hand if possible. returns 1 on success.
-/mob/proc/put_in_inactive_hand(var/obj/item/W)
-	return put_in_hand(get_inactive_hand_index(), W)
+/mob/proc/put_in_inactive_hand(var/obj/item/W) // The technology has advanced, we can now handle lifeforms with more than 2 hands.
+	for (var/i = get_inactive_hand_index(); i > 0; i--) // Get inactive hand index returns us to the last available hand. We iterate until we come to 0.
```

Comment:
From what I can tell, `get_inactive_hand_index()` returns the last available hand *only* if `active_hand - 1 < 1`, which I don't think will always be the case. Why not set `i` to `held_items.len` directly?

Reply:
I had not thought of that. When I tested it locally, debug showed that `get_inactive_hand_index()` and `held_item.len` always coincided. I guess I chose the wrong var.

Reply:
Did you test with a multi-handed mob, with various hands being the active hand?

Reply:
Yes. I tested with a Martian.

Reply:
Strange. If it works, it works, but I'm not sure *why* it works.
Also, did you make sure that this loop excludes the active hand?

Reply:
AFAIK it doesn't excludes the active hand. However, if the active hand is available, it is chosen first. If it isn't available and we iterate through it, we will fail the check.

Reply:
That should work fine, though it would be `!=`, not `=!`.